### PR TITLE
FAI-250: [trusty-ui]: Move to kogito-tooling 0.6.1

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-250

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: ui-packages/packages/trusty/src/envelope/envelope.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  bus: {
+    postMessage<D, Type>(
+      message: EnvelopeBusMessage<D, Type>,
+      targetOrigin?: string,
+      _?: any
+    ) {
+      window.parent.postMessage(message, '*', _);
+    }
+  },
+  editorFactory: new EditorEnvelope.CompositeEditorFactory([
```
Comment (manstis) [inline ui-packages/packages/trusty/src/envelope/envelope.ts]: I should be able to copy the envelope HTML and JS from `@kogito-tooling/kie-bc-editors`.

I was going to add the PMML editor factory here too; but it's probably best I make a quick and easy "host" for the new editor until it's at a stage to be re-used in kogito's channels (including trusty). I'll update this PR... converting to _draft_.

**Comment by manstis** [comment]
Jenkins retest this

File: ui-packages/packages/trusty/src/components/Templates/TrustyApp/TrustyApp.tsx [hunk 2]
```diff
@@ -17,13 +10,24 @@ import {
...
 import kogitoLogo from '../../../../static/images/kogitoLogo.svg';
-import AuditDetail from '../AuditDetail/AuditDetail';
 import imgAvatar from '../../../../static/images/user.svg';
+import { I18n } from "../../../i18n/I18n";
 import Breadcrumbs from '../../Organisms/Breadcrumbs/Breadcrumbs';
+import AuditDetail from '../AuditDetail/AuditDetail';
+import AuditOverview from '../AuditOverview/AuditOverview';
 import './TrustyApp.scss';
 
+export const I18nContext = React.createContext<I18nContextType<I18n>>({} as any);
```
Comment (kelvah) [inline ui-packages/packages/trusty/src/components/Templates/TrustyApp/TrustyApp.tsx]: I see you're setting up a context for i18n using the setup provided by `kogito-tooling`.

I suppose the idea is to use it for the whole trusty app? I thought i18n could deserve a separate PR. Unless you cannot do without it with the editors.

File: ui-packages/packages/trusty/webpack.common.js [hunk 3]
```diff
@@ -15,7 +15,8 @@ module.exports = {
   plugins: [
     new HtmlWebpackPlugin({
       template: path.resolve(__dirname, 'src', 'index.html'),
-      favicon: 'src/favicon.ico'
+      favicon: 'src/favicon.ico',
+      chunks: ['app']
```
Comment (kelvah) [inline ui-packages/packages/trusty/webpack.common.js]: Why are you adding chunks configuration?

File: ui-packages/packages/trusty/src/components/Templates/TrustyApp/TrustyApp.tsx [hunk 4]
```diff
@@ -1,11 +1,4 @@
-import React, { useState } from 'react';
-import {
-  NavLink,
-  Redirect,
-  Route,
-  Switch,
-  useLocation
-} from 'react-router-dom';
+import { I18nContextType } from "@kogito-tooling/i18n";
```
Comment (kelvah) [inline ui-packages/packages/trusty/src/components/Templates/TrustyApp/TrustyApp.tsx]: I think it's a good idea if we use some kind of "rules" for sorting imports. Like app-wide requirements on top (react, routing etc), external components (patternfly or similar) and internal stuff (ui components, types etc.).

What do you think? (I'm not saying I always ordered things properly but I try :grimacing:)

I don't know if you changed the order manually of if it was your IDE.
I think there are also ESLint rules that auto sort imports that we could use once we migrate to it.

**Review by kelvah — COMMENTED** [review]
Hi @manstis, I left some questions.

I also think the files you changed are not formatted by Prettier. Is it possible?

File: ui-packages/packages/trusty/src/envelope/envelope.ts (hunk 1 above)
Comment (manstis) [reply ui-packages/packages/trusty/src/envelope/envelope.ts]: Done.. ignore this comment.

File: ui-packages/packages/trusty/src/components/Templates/TrustyApp/TrustyApp.tsx (hunk 2 above)
Comment (manstis) [reply ui-packages/packages/trusty/src/components/Templates/TrustyApp/TrustyApp.tsx]: I agree.. I was originally adding the basics to support i18n (using `kogito-tooling`) but it was turning into much more than I anticipated. I have now removed everything from this PR relating to i18n.

File: ui-packages/packages/trusty/webpack.common.js (hunk 3 above)
Comment (manstis) [reply ui-packages/packages/trusty/webpack.common.js]: Well...... when I add back a custom `envelope.ts` to include the PMML editor factory I need an additional `entry` in `webpack.common.js` to compile `envelope/envelope.ts` as a separate chunk to use soley from the IFRAME (and not in the main `app` chunk).. I was originally excluding `envelope` but thought it less error prone to specify what to include. However since I am now using the `envelope.ts` from a dependency and only have one chunk I can remove this...

File: ui-packages/packages/trusty/src/components/Templates/TrustyApp/TrustyApp.tsx (hunk 4 above)
Comment (manstis) [reply ui-packages/packages/trusty/src/components/Templates/TrustyApp/TrustyApp.tsx]: I am formatting the imports according to the rules I have setup in my IDE.

IIRC I acquired them from `kogito-tooling`.. but I would need to check where/how they are configured now..

But I agree with you re: standardising to avoid this type of white space commit.

**Review by kelvah — APPROVED** [review]
Thank you @manstis. Looks fine.

Could you just check if Prettier was run upon the changes?

**Comment by manstis** [comment]
@kelvah Yes I have https://github.com/prettier/prettier-vscode installed and enabled to format files on save.

**Review by jiripetrlik — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]
